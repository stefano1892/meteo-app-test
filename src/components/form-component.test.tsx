import { render } from "@testing-library/react"
import FormComponent from "./form-component"
import userEvent from "@testing-library/user-event"

describe('FormComponent', () => {
    test('Submit button is disabled', () => {
        const { asFragment } = render(
            <FormComponent 
                handleInputChange={jest.fn()} 
                handleSubmit={jest.fn()}
                city=""
            />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('Submit button is enabled', () => {
        const { asFragment } = render(
            <FormComponent 
                handleInputChange={jest.fn()} 
                handleSubmit={jest.fn()}
                city="Milan"
            />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('HandleInputChange works', async () => {
        const handleInputChange = jest.fn()
        const { getByPlaceholderText } = render(
            <FormComponent 
                handleInputChange={handleInputChange}
                handleSubmit={jest.fn()}
                city="Milan"
            />
        )

        const cityToSearch = 'Milan'

        const inputElement = getByPlaceholderText('Inserisci la città...')
        await userEvent.type(inputElement, cityToSearch)
        expect(handleInputChange).toHaveBeenCalledTimes(cityToSearch.length)
    })

    test('HandleSubmit works', async () => {
        const handleSubmit = jest.fn()
        const {getByPlaceholderText, getByRole} = render(
            <FormComponent 
                handleSubmit={handleSubmit}
                handleInputChange={jest.fn()}
                city="Milan"
            />
        )
        
        const cityToSearch = 'Milan'

        const inputElement = getByPlaceholderText('Inserisci la città...')
        await userEvent.type(inputElement, cityToSearch)

        const buttonElement = getByRole('button', {name: 'Cerca'})
        await userEvent.click(buttonElement)

        expect(handleSubmit).toHaveBeenCalledTimes(1)
    })
})