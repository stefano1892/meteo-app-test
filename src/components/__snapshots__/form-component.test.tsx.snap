// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormComponent Submit button is disabled 1`] = `
<DocumentFragment>
  <form>
    <div
      class="form-container"
    >
      <input
        class="cityInput"
        placeholder="Inserisci la città..."
        type="text"
        value=""
      />
      <button
        class="btn btn-submit"
        disabled=""
        type="submit"
      >
        Cerca
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`FormComponent Submit button is enabled 1`] = `
<DocumentFragment>
  <form>
    <div
      class="form-container"
    >
      <input
        class="cityInput"
        placeholder="Inserisci la città..."
        type="text"
        value="Milan"
      />
      <button
        class="btn btn-submit"
        type="submit"
      >
        Cerca
      </button>
    </div>
  </form>
</DocumentFragment>
`;
